### ISSUE TITLE
MPI_Comm_spawn(): Warnings about failing handshake

### ISSUE THREAD

**Issue by dalcinl** [open-mpi/ompi#4130]
This is from mpi4py tests involving `MPI_Comm_spawn()`. I getting the output below, but not consistently. I've not been able to reproduce the issue with simple pure-C example.

Any tips about how to further debug this issue?

```
[kw14821][[15990,32],0][btl_tcp.c:556:mca_btl_tcp_recv_blocking] remote peer unexpectedly closed connection while I was waiting for blocking message
--------------------------------------------------------------------------
WARNING: Open MPI failed to handshake with a connecting peer MPI
process over TCP.  This should not happen.

Your Open MPI job may now fail.

  Local host: kw14821
  PID:        33504
  Message:    did not receive entire connect ACK from peer
--------------------------------------------------------------------------
```

**Comment by dalcinl** [comment]
Maybe a duplicate of #4131.

**Comment by ggouaillardet** [comment]
this is a duplicate of #4131 indeed (and i made a tentative fix in #4134), so let's close this issue for now.
